core/src/main/scala/org/apache/spark/deploy/worker/DriverWrapper.scala

```diff
@@ -38,8 +39,10 @@ object DriverWrapper {
        */
       case workerUrl :: userJar :: mainClass :: extraArgs =>
         val conf = new SparkConf()
-        val rpcEnv = RpcEnv.create("Driver",
-          Utils.localHostName(), 0, conf, new SecurityManager(conf))
+        val host: String = Utils.localHostName()
```

Comment:
Please correct me, AFAIU `spark.driver.host` may not be specified on each node of the cluster and its value might be global(i.e. cluster wide). Since driver is launched on a random node during a failover, it's value can not be pre-assigned and has to be picked up from that node's local environment.

About `spark.driver.port`, I am not sure, but the fact is - it might work, even if it is global cluster wide constant.